### ISSUE TITLE
Make on sale prices accessibile

### ISSUE THREAD

**Issue by gigitux** [woothemes/woocommerce#31099]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**
WooCommerce renders sale products with the old price wrapped by `<del>` tags and the current price next to it.
Currently, screen readers read both prices without specifying which one is the old one and which one is the current one.

**Expected behavior**:
Ideally, the screen reader should read something along these lines:
```
Previous price: 123$, Discounted price: 100$
```

More details on https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/1668

**Comment by juliaamosova** [comment]
Hi @gigitux,

Thank you for opening the issue! It requires further feedback from the WooCommerce Core team. I am adding the `needs developer feedback` label to this issue so that the Core team could take a look. 

Please note it may take a few days for them to get to this issue. Thank you for your patience.

**Comment by discoinfiltrator** [comment]
> Currently, screen readers read both prices without specifying which one is the old one and which one is the current one.

This was true of `woocommerce-blocks`, but here the original price is being hidden because of a bad decision in https://github.com/woocommerce/woocommerce/issues/29379

I've put in a PR to align this code with the approach made in `woocommerce-blocks`.

**Comment by coreymckrill** [comment]
Possibly related info: https://www.webaxe.org/strikethrough-html-accessibility/

**Comment by BFTrick** [comment]
We also had this issue come up in an accessibility audit.

If you look into [WCAG 1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html) - if you're presenting information to sighted users. You need to present that same information to screen readers. 

So yes. This is a requirement.

Here's a quick demo with screen reader software to see the problem (🔉 on):

https://github.com/woocommerce/woocommerce/assets/1065372/0ab7bda7-9df4-4397-a25c-857b952fcc44

If you look at the big players they're already accessible. 👇 

## Amazon

Amazon has the discount savings, the current price, and the original price (list price).

Here's what's read to screen readers:

> $219.99 with 49 percent savings

> List Price: $429.99

![Screenshot 2024-02-06 at 11 19 38 AM](https://github.com/woocommerce/woocommerce/assets/1065372/814754b2-ba7e-4405-8ccd-9c321b9307a2)


``` html
<div id="corePriceDisplay_desktop_feature_div" class="celwidget" data-feature-name="corePriceDisplay_desktop" data-csa-c-type="widget" data-csa-c-content-id="corePriceDisplay_desktop" data-csa-c-slot-id="corePriceDisplay_desktop_feature_div" data-csa-c-asin="B09WYYW4TR" data-csa-c-is-in-initial-active-row="false" data-csa-c-id="2vhfkv-2d2gah-1bh3oz-7ahxkd" data-cel-widget="corePriceDisplay_desktop_feature_div">
    <div class="a-section a-spacing-none aok-align-center aok-relative">
        <span class="aok-offscreen">$219.99 with 49 percent savings</span>
        <span aria-hidden="true" class="a-size-large a-color-price savingPriceOverride aok-align-center reinventPriceSavingsPercentageMargin savingsPercentage">-49%</span>
        <span class="a-price aok-align-center reinventPricePriceToPayMargin priceToPay" data-a-size="xl" data-a-color="base">
            <span class="a-offscreen"></span>
            <span aria-hidden="true">
                <span class="a-price-symbol">$</span>
                <span class="a-price-whole">219<span class="a-price-decimal">.</span></span>
                <span class="a-price-fraction">99</span>
            </span>
        </span>
        <span id="taxInclusiveMessage" class="a-size-mini a-color-base aok-align-center aok-nowrap"></span>
    </div>
    <div class="a-section a-spacing-small aok-align-center">
        <span>
            <span class="aok-relative">
                <span class="a-size-small aok-offscreen">List Price: $429.99</span>
                <span aria-hidden="true" class="a-size-small a-color-secondary aok-align-center basisPrice">List Price:
                    <span class="a-price a-text-price" data-a-size="s" data-a-strike="true" data-a-color="secondary">
                        <span class="a-offscreen">$429.99</span>
                        <span aria-hidden="true">$429.99</span>
                    </span>
                </span>
            </span>
            <span class="a-size-small aok-align-center basisPriceLegalMessage">
                <span class="a-declarative" data-action="a-popover" data-csa-c-type="widget" data-csa-c-func-deps="aui-da-a-popover" data-a-popover="{&quot;closeButton&quot;:&quot;true&quot;,&quot;name&quot;:&quot;basisPriceLegalMessageDisplayPreload-321bac8b-9856-4418-ba73-3ac55debec86&quot;,&quot;position&quot;:&quot;triggerBottom&quot;}" data-csa-c-id="ww7fnj-urvbi3-gcbxhg-6u22y0">
                    <a aria-label="Learn more about Amazon pricing and savings" href="javascript:void(0)" role="button" class="a-popover-trigger a-declarative reinventPriceLegalMessagePopover">
                        <svg aria-hidden="true" class="reinventPrice_legalMessage_icon" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
                            <path d="M256,9C119,9,8,120.08,8,257S119,505,256,505,504,394,504,257,393,9,256,9Zm0,76.31A47.69,47.69,0,1,1,208.31,133,47.69,47.69,0,0,1,256,85.31Zm38.15,332.38a12.18,12.18,0,0,1-12.21,12H229.67a11.85,11.85,0,0,1-11.82-12V249.92a11.86,11.86,0,0,1,11.82-12h52.27a12.18,12.18,0,0,1,12.21,12Z"></path>
                        </svg>
                        <i class="a-icon a-icon-popover"></i>
                    </a>
                </span>
                <div class="a-popover-preload" id="a-popover-basisPriceLegalMessageDisplayPreload-321bac8b-9856-4418-ba73-3ac55debec86">
                    <div class="a-section a-spacing-none">
                        <h3></h3>
                        <span class="a-size-base">The List Price is the suggested retail price of a new product as provided by a manufacturer, supplier, or seller. Except for books, Amazon will display a List Price if the product was purchased by customers on Amazon or offered by other retailers at or above the List Price in at least the past 90 days. List prices may not necessarily reflect the product's prevailing market price.
                            <br>
                            <a class="a-link-normal" href="https://www.amazon.com/gp/help/customer/display.html?nodeId=GQ6B6RH72AX8D2TD&amp;ref_=dp_hp">Learn more</a>
                        </span>
                    </div>
                </div>
            </span>
        </span>
    </div>
</div>
```

## Walmart

![Screenshot 2024-02-06 at 11 26 53 AM](https://github.com/woocommerce/woocommerce/assets/1065372/b53dfba6-bf78-4f23-8392-487fe04ca989)

Walmart also lists the current price, how much you save, and the original price

> Now $16.99

> You save $10.00

> Was $26.99

``` html
<span data-testid="price-wrap" class="b lh-copy dark-gray f1 mr2 green">
  <span hidden="" itemprop="priceCurrency">USD</span>
  <span class="inline-flex flex-column">
    <span itemprop="price" aria-hidden="false">Now $16.99</span>
  </span>
</span>
<span class="w_iUH7">
  <div>You save $10.00</div>
</span>
<div class="nowrap items-center inline-flex">
  <span class="w_iUH7">was $26.99</span>
  <span aria-hidden="true" class="mr2 f6 gray strike">$26.99</span>
  <button class="w_hhLG w_DZvO w_0_LY nl1 pa0 mr2 no-underline v-btm h3" type="button" data-pcss-hide="true" aria-label="learn more about strikethrough prices">
    <i class="ld ld-InfoCircle" style="font-size: 1rem; vertical-align: -0.175em; width: 1rem; height: 1rem; box-sizing: content-box;"></i>
  </button>
</div>
<div style="flex-basis: 100%; height: 0px;"></div>
<div class="flex items-center" aria-hidden="true" data-testid="dollar-saving">
  <span class="bg-washed-green green ph1 mr1 b br1 nowrap f6 lh-copy" style="padding-top: 2px; padding-bottom: 2px;">You save</span>
  <span class="green b lh-copy f5">$10.00</span>
</div>
```
---

Here's what I'd propose for WooCommerce:

``` html
  <del>
    <span class="screen-reader-text">Original price was:</span>
    $79.99
  </del> 
  <ins>
    <span class="screen-reader-text">Current price is:</span>
    $23.99
  </ins>
```

It works well with the current markup and it uses the `screen-reader-text` class which is already incorporated into WordPress core.

**Comment by jorgeatorres** [comment]
Fixed in #44413.
